diff -ruN bochs-2.8/config.sub bochs-2.8-banan_os/config.sub
--- bochs-2.8/config.sub	2024-03-10 08:59:37.000000000 +0200
+++ bochs-2.8-banan_os/config.sub	2024-12-07 05:15:32.948901314 +0200
@@ -1368,7 +1368,7 @@
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
-	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi*)
+	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* | banan_os*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
