diff -ruN freetype-2.13.3/builds/unix/configure freetype-2.13.3-banan_os/builds/unix/configure
--- freetype-2.13.3/builds/unix/configure	2024-08-12 14:00:49.000000000 +0300
+++ freetype-2.13.3-banan_os/builds/unix/configure	2025-07-31 17:40:12.168493444 +0300
@@ -5474,6 +5474,10 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+banan_os*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 beos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
@@ -11261,6 +11265,16 @@
   esac
   ;;
 
+banan_os*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  dynamic_linker="$host_os DynamicLoader.so"
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
 beos*)
   library_names_spec='$libname$shared_ext'
   dynamic_linker="$host_os ld.so"
