diff -ruN openssl-3.3.1/crypto/bio/bss_dgram.c openssl-3.3.1-banan_os/crypto/bio/bss_dgram.c
--- openssl-3.3.1/crypto/bio/bss_dgram.c	2024-06-04 15:53:04.000000000 +0300
+++ openssl-3.3.1-banan_os/crypto/bio/bss_dgram.c	2025-06-01 19:48:55.088806701 +0300
@@ -61,7 +61,7 @@
 #   define NO_RECVMMSG
 #  endif
 # endif
-# if defined(__GNU__)
+# if defined(__GNU__) || defined(__banan_os__)
    /* GNU/Hurd does not have IP_PKTINFO yet */
    #undef NO_RECVMSG
    #define NO_RECVMSG
