diff -ru openssh-10.0p1/auth-passwd.c openssh-10.0p1-banan_os/auth-passwd.c
--- openssh-10.0p1/auth-passwd.c	2025-04-09 10:02:43.000000000 +0300
+++ openssh-10.0p1-banan_os/auth-passwd.c	2025-08-16 21:36:13.372676511 +0300
@@ -83,6 +83,11 @@
 	static int expire_checked = 0;
 #endif
 
+#ifdef __banan_os__
+	debug_f("bypassing password auth on banan-os");
+	return 1;
+#endif
+
 	if (strlen(password) > MAX_PASSWORD_LEN)
 		return 0;
 
