diff -ru openssh-10.0p1/session.c openssh-10.0p1-banan_os/session.c
--- openssh-10.0p1/session.c	2025-04-09 10:02:43.000000000 +0300
+++ openssh-10.0p1-banan_os/session.c	2025-08-16 20:33:52.515584804 +0300
@@ -1039,11 +1039,13 @@
 # endif /* HAVE_CYGWIN */
 #endif /* HAVE_LOGIN_CAP */
 
+#ifdef _PATH_MAILDIR
 	if (!options.use_pam) {
 		snprintf(buf, sizeof buf, "%.200s/%.50s",
 		    _PATH_MAILDIR, pw->pw_name);
 		child_set_env(&env, &envsize, "MAIL", buf);
 	}
+#endif
 
 	/* Normal systems set SHELL by default. */
 	child_set_env(&env, &envsize, "SHELL", shell);
