diff -ruN curl-8.11.1/configure curl-8.11.1-banan_os/configure
--- curl-8.11.1/configure	2024-12-11 09:08:12.000000000 +0200
+++ curl-8.11.1-banan_os/configure	2025-01-22 18:29:34.167111729 +0200
@@ -15537,6 +15537,16 @@
   esac
   ;;
 
+banan_os*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  dynamic_linker="$host_os DynamicLoader.so"
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
 beos*)
   library_names_spec='$libname$shared_ext'
   dynamic_linker="$host_os ld.so"
