diff -ruN gmp-6.3.0/configfsf.sub gmp-6.3.0-banan_os/configfsf.sub
--- gmp-6.3.0/configfsf.sub	2023-07-29 16:42:16.000000000 +0300
+++ gmp-6.3.0-banan_os/configfsf.sub	2024-08-07 21:37:12.147551767 +0300
@@ -1759,7 +1759,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* )
+	     | fiwix* | mlibc* | banan_os* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
