diff -ruN libogg-1.3.6/configure libogg-1.3.6-banan_os/configure
--- libogg-1.3.6/configure	2025-06-16 19:02:25.000000000 +0300
+++ libogg-1.3.6-banan_os/configure	2025-10-31 22:25:25.050050235 +0200
@@ -10622,6 +10622,16 @@
   esac
   ;;
 
+banan_os*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  dynamic_linker="$host_os DynamicLoader.so"
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
 beos*)
   library_names_spec='$libname$shared_ext'
   dynamic_linker="$host_os ld.so"
