diff -ruN libX11-1.8.12/configure libX11-1.8.12-banan_os/configure
--- libX11-1.8.12/configure	2025-03-09 01:53:09.000000000 +0200
+++ libX11-1.8.12-banan_os/configure	2025-11-13 20:07:46.592619585 +0200
@@ -6633,6 +6633,10 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+banan_os*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 beos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
@@ -12138,6 +12142,16 @@
   esac
   ;;
 
+banan_os*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  dynamic_linker="$host_os DynamicLoader.so"
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
 beos*)
   library_names_spec='$libname$shared_ext'
   dynamic_linker="$host_os ld.so"
