diff -ruN libuv-v1.51.0/src/unix/udp.c libuv-1.51.0-x86_64/src/unix/udp.c
--- libuv-v1.51.0/src/unix/udp.c	2025-04-25 12:50:27.000000000 +0300
+++ libuv-1.51.0-x86_64/src/unix/udp.c	2025-07-23 17:15:38.986952773 +0300
@@ -767,7 +767,8 @@
     !defined(__ANDROID__) &&                                        \
     !defined(__DragonFly__) &&                                      \
     !defined(__QNX__) &&                                            \
-    !defined(__GNU__)
+    !defined(__GNU__) &&                                            \
+    !defined(__banan_os__)
 static int uv__udp_set_source_membership4(uv_udp_t* handle,
                                           const struct sockaddr_in* multicast_addr,
                                           const char* interface_addr,
@@ -958,7 +959,8 @@
     !defined(__ANDROID__) &&                                        \
     !defined(__DragonFly__) &&                                      \
     !defined(__QNX__) &&                                            \
-    !defined(__GNU__)
+    !defined(__GNU__) &&                                            \
+    !defined(__banan_os__)
   int err;
   union uv__sockaddr mcast_addr;
   union uv__sockaddr src_addr;
