diff -ruN nano-8.5/lib/getprogname.c nano-8.5-banan_os/lib/getprogname.c
--- nano-8.5/lib/getprogname.c	2025-06-12 10:29:39.000000000 +0300
+++ nano-8.5-banan_os/lib/getprogname.c	2025-08-19 01:25:56.020428849 +0300
@@ -50,7 +50,7 @@
 # include <sys/procfs.h>
 #endif
 
-#if defined __SCO_VERSION__ || defined __sysv5__
+#if defined __SCO_VERSION__ || defined __sysv5__ || defined __banan_os__
 # include <fcntl.h>
 # include <string.h>
 #endif
@@ -265,7 +265,7 @@
         }
     }
   return NULL;
-# elif defined __SCO_VERSION__ || defined __sysv5__                /* SCO OpenServer6/UnixWare */
+# elif defined __SCO_VERSION__ || defined __sysv5__ || defined __banan_os__                /* SCO OpenServer6/UnixWare */
   char buf[80];
   int fd;
   sprintf (buf, "/proc/%d/cmdline", getpid());
