diff -ruN lynx2.9.2/config.sub lynx2.9.2-banan_os/config.sub
--- lynx2.9.2/config.sub	2023-12-27 16:41:27.000000000 +0200
+++ lynx2.9.2-banan_os/config.sub	2024-08-21 14:23:47.519364107 +0300
@@ -1768,7 +1768,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* | cos* | mbr* | ironclad* )
+	     | fiwix* | mlibc* | cos* | mbr* | ironclad* | banan_os* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
