diff -ruN sdl12-compat-release-1.2.68/src/SDL12_compat.c sdl12-compat-release-1.2.68-banan_os/src/SDL12_compat.c
--- sdl12-compat-release-1.2.68/src/SDL12_compat.c	2023-09-26 20:43:48.000000000 +0300
+++ sdl12-compat-release-1.2.68-banan_os/src/SDL12_compat.c	2025-06-26 03:40:42.751597384 +0300
@@ -1201,7 +1201,11 @@
     }
 #elif defined(__unix__)
     #include <dlfcn.h>
+# if defined(__banan_os__)
+    #define SDL20_LIBNAME "libSDL2.so"
+# else
     #define SDL20_LIBNAME "libSDL2-2.0.so.0"
+# endif
     #define SDL20_REQUIRED_VER SDL_VERSIONNUM(2,0,7)
     static void *Loaded_SDL20 = NULL;
     #define LoadSDL20Library() ((Loaded_SDL20 = dlopen(SDL20_LIBNAME, RTLD_LOCAL|RTLD_NOW)) != NULL)
