diff -ru mesa-25.0.7-banan_os/src/gallium/drivers/llvmpipe/lp_texture.c mesa-25.0.7-x86_64/src/gallium/drivers/llvmpipe/lp_texture.c
--- mesa-25.0.7-banan_os/src/gallium/drivers/llvmpipe/lp_texture.c      2025-05-28 18:20:23.000000000 +0300
+++ mesa-25.0.7-x86_64/src/gallium/drivers/llvmpipe/lp_texture.c        2025-08-03 19:16:20.254971098 +0300
@@ -59,7 +59,7 @@
 #include "frontend/sw_winsys.h"
 #include "git_sha1.h"
 
-#ifndef _WIN32
+#if !defined(_WIN32) && !defined(__banan_os__)
 #include "drm-uapi/drm_fourcc.h"
 #endif
 
@@ -1713,7 +1713,7 @@
    case PIPE_RESOURCE_PARAM_LAYER_STRIDE:
       *value = lpr->img_stride[level];
       return true;
-#ifndef _WIN32
+#if !defined(_WIN32) && !defined(__banan_os__)
    case PIPE_RESOURCE_PARAM_MODIFIER:
       *value = lpr->dmabuf ? DRM_FORMAT_MOD_LINEAR : DRM_FORMAT_MOD_INVALID;
       return true;
