From a4e6b807885a7cb4a507b4e114743aa0004376ad Mon Sep 17 00:00:00 2001
From: Oskari Alaranta <oskari.alaranta@bananymous.com>
Date: Sun, 10 Aug 2025 01:57:38 +0300
Subject: [PATCH] add support for custom SDL path

---
 doomgeneric/Makefile.sdl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/doomgeneric/Makefile.sdl b/doomgeneric/Makefile.sdl
index 38402da..8c2f84e 100644
--- a/doomgeneric/Makefile.sdl
+++ b/doomgeneric/Makefile.sdl
@@ -12,8 +12,8 @@ else
 endif
 
 
-SDL_CFLAGS = `sdl2-config --cflags`
-SDL_LIBS = `sdl2-config --cflags --libs` -lSDL2_mixer
+SDL_CFLAGS = `$(SDL_PATH)sdl2-config --cflags`
+SDL_LIBS = `$(SDL_PATH)sdl2-config --cflags --libs` -lSDL2_mixer
 
 
 CC=clang  # gcc or g++
-- 
2.50.1

